### ISSUE TITLE
Unexpected errors in Python 3.14: attr-defined, arg-type, misc

### ISSUE THREAD

**Issue by Tatsh** [python/mypy#20186]
**Bug Report**

I am getting unexpected errors with my [portage-stubs](https://github.com/Tatsh/portage-stubs/tree/master/portage-stubs) package when I use Mypy with Python 3.14, even with the latest version of Mypy. It is working fine with 3.13 and below.

[Workflow run](https://github.com/Tatsh/livecheck/actions/runs/19111769601/job/54610456912#step:8:12)

**To Reproduce**

```shell
python3.14 -m venv mypy314-bug
. mypy314-bug/bin/activate
pip install git+https://github.com/python/mypy.git git+https://github.com/gentoo/portage.git portage-stubs mypy

cat > test.py <<EOF
from portage.versions import catpkgsplit
import portage

split_or_none = catpkgsplit('dev-lang/python')
assert split_or_none is not None
r0, r1, r2, r3 = split_or_none

settings = portage.config(clone=portage.settings)

portage.doebuild('a-path', 'unpack', settings=settings, tree='porttree')
EOF

mypy test.py
```

**Expected Behavior**

It should say `Success: no issues found in 1 source file`.

**Actual Behavior**

```plain
test.py:6: error: Need more than 2 values to unpack (4 expected)  [misc]
test.py:6: error: Need more than 3 values to unpack (4 expected)  [misc]
test.py:8: error: Module has no attribute "config"  [attr-defined]
test.py:10: error: Argument 2 to "doebuild" has incompatible type "Literal['unpack']"; expected "Literal['info', 'nofetch', 'pretend', 'configure', 'compile', 'merge', 'test', 'install', 'prepare', 'clean', 'cleanrm', 'depend', 'digest', 'fetch', 'fetchall', 'help', 'manifest', 'qmerge', 'postinst', 'postrm', 'preinst', 'prerm', 'setup']"  [arg-type]
```

It does not see the last literal value `'unpack'`. It thinks `catpkgsplit` returns a 2-item tuple. It does not think `portage.config` exists.

**Your Environment**

- Mypy version used: HEAD
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.14

**Comment by sterliakov** [comment]
Okay, here's your clue:

```
$ python3.14 -m venv .venv
$ . .venv/bin/activate
$ pip install portage-stubs
Collecting portage-stubs
  Using cached portage_stubs-0.0.3-py3-none-any.whl.metadata (1.2 kB)
Using cached portage_stubs-0.0.3-py3-none-any.whl (3.8 kB)
Installing collected packages: portage-stubs
Successfully installed portage-stubs-0.0.3
```

Oops. Check your package metadata, 0.0.3 is installed instead of 0.0.6 (latest as of now) on py3.14.

The same thing with that workflow, here's their lockfile:

https://github.com/Tatsh/livecheck/blob/3fd955442b791c3abab6fcd04f7eaa9a327cd4e8/poetry.lock#L1777-L1802

That's all because of this line that you have fixed yesterday:

https://github.com/Tatsh/portage-stubs/commit/45352dea5040af13d880df4a830b80e411157102#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L69-L71

But the latest release (0.0.6) still has that `<3.14` pin. Cut a `0.0.7` release and this should go away, there's no mypy fault in installing an old version of a stub package.

Installing on 3.13 confirms that (look at line 9):

```
$ head .venv/lib/python3.13/site-packages/portage_stubs-0.0.6.dist-info/METADATA 
Metadata-Version: 2.3
Name: portage-stubs
Version: 0.0.6
Summary: PEP 561 type stubs for portage.
License: MIT
Keywords: gentoo,pep561,portage,stubs,types
Author: Andrew Udvare
Author-email: audvare@gmail.com
Requires-Python: >=3.10,<3.14
Classifier: Development Status :: 2 - Pre-Alpha
```
